shenyu-admin/src/main/java/org/apache/shenyu/admin/discovery/DiscoveryDataChangedEventSyncListener.java

```diff
@@ -0,0 +1,148 @@
+        List<DiscoveryUpstreamData> upstreamDataList = discoverySyncData.getUpstreamDataList();
+        if (needPersistence) {
+            if (CollectionUtils.isEmpty(upstreamDataList)) {
+                LOGGER.warn("shenyu proxySelectorData#discoveryUpstreamList is empty");
+                return;
+            }
+            switch (currentEvent) {
+                case ADDED:
+                    upstreamDataList.forEach(d -> {
+                        DiscoveryUpstreamDO discoveryUpstreamDO = new DiscoveryUpstreamDO();
+                        BeanUtils.copyProperties(d, discoveryUpstreamDO);
+                        discoveryUpstreamDO.setId(UUIDUtils.getInstance().generateShortUuid());
+                        discoveryUpstreamDO.setDateCreated(new Timestamp(System.currentTimeMillis()));
+                        discoveryUpstreamDO.setDateUpdated(new Timestamp(System.currentTimeMillis()));
+                        discoveryUpstreamMapper.insert(discoveryUpstreamDO);
+                        LOGGER.info("shenyu [DiscoveryDataChangedEventSyncListener] ADDED Upstream {}", discoveryUpstreamDO.getUrl());
+                    });
+                    fillFullyDiscoverySyncData(discoverySyncData);
+                    dataChangedEvent = new org.apache.shenyu.admin.listener.DataChangedEvent(ConfigGroupEnum.PROXY_SELECTOR, DataEventTypeEnum.CREATE, Collections.singletonList(discoverySyncData));
+                    break;
+                case UPDATED:
+                    upstreamDataList.forEach(d -> {
+                        DiscoveryUpstreamDO discoveryUpstreamDO = new DiscoveryUpstreamDO();
+                        BeanUtils.copyProperties(d, discoveryUpstreamDO);
+                        discoveryUpstreamMapper.update(discoveryUpstreamDO);
+                        LOGGER.info("shenyu [DiscoveryDataChangedEventSyncListener] UPDATE Upstream {}", discoveryUpstreamDO.getUrl());
+                    });
+                    fillFullyDiscoverySyncData(discoverySyncData);
+                    dataChangedEvent = new org.apache.shenyu.admin.listener.DataChangedEvent(ConfigGroupEnum.PROXY_SELECTOR, DataEventTypeEnum.UPDATE, Collections.singletonList(discoverySyncData));
+                    break;
+                case DELETED:
+                    if (CollectionUtils.isNotEmpty(upstreamDataList)) {
+                        upstreamDataList.forEach(up -> {
+                            discoveryUpstreamMapper.deleteByUrl(up.getUrl());
+                            LOGGER.info("shenyu [DiscoveryDataChangedEventSyncListener] DELETE Upstream {}", up.getUrl());
+                        });
+                    }
+                    fillFullyDiscoverySyncData(discoverySyncData);
+                    dataChangedEvent = new org.apache.shenyu.admin.listener.DataChangedEvent(ConfigGroupEnum.PROXY_SELECTOR, DataEventTypeEnum.UPDATE, Collections.singletonList(discoverySyncData));
+                    break;
+                default:
```

Comment:
not have default, or throw illegalException

Reply:
done